src/matchers/strings.lua

```diff
@@ -1,4 +1,22 @@
+local prettyValue = require("src.utils.prettyValue")
+local assertType = require("src.asserts.type")
+
 ---@type lest.Matcher
-local function toMatch(ctx, received, pattern) end
+local function toMatch(ctx, received, pattern)
+	assertType(received, "string")
```

Comment:
This should not be done on any matcher's received value. If the received value is invalid it should return a failing matcher result. We use errors for the matcher args because those are defined in the test, whereas the received value comes from the project's actual source code.